### ISSUE TITLE
Quickly losing its connection to the DB host

### ISSUE THREAD

**Issue by chw-webops** [Sequel-Ace/Sequel-Ace#786]
Version of Sequel-Ace? v3.1.0 (build 3013)
Version of macOS? Catalina, 10.15.7
Device(s): Macbook pro
Installed via homebrew (cask)

Issue: Sequel Ace is very quickly losing its connection to the DB host.

I connect using SSH method. If I open a database table to view its contents and leave the connection idle for about 20 to 30 seconds I get a dialog that pops up that indicates the following: "Connection Lost - Sequel Ace appear to have lost the connection to the server, or the server has stopped responding", with a choice to close the connection or reconnect.

If I choose reconnect, it attempts to connect and reload the database table I was looking at. But the dialog pops up again with the same message and choices. The only way to move forward is to close the connection and open a new one.

This behavior began after updating to v3.1.0 from v2.3.2. On the 2-series release this wasn't an issue.

I've toggled the Keep Alive option under Network preferences and also adjusted the connection timeout. Nothing seems to resolve the issue (even setting the timeout value to zero "0" with keep-alive enabled.)

**Comment by Kaspik** [comment]
If you upgraded directly from 2 version - could you backup favorites, wipe Sequel via AppCleaner (for example) and install fresh new? There are some issues with non-ARC app migration to ARC version in 3.0.0...

The popup is known issue fixed today via https://github.com/Sequel-Ace/Sequel-Ace/pull/772 and kind of duplicate of https://github.com/Sequel-Ace/Sequel-Ace/issues/658

We will have beta ready soon.

**Comment by chw-webops** [comment]
Sure, I'll give that a try and let you know. Thanks!

**Comment by chw-webops** [comment]
@Kaspik Wiping Sequel Ace and reinstalling fresh didn't seem to work.

**Comment by Kaspik** [comment]
Thanks for the info.

**Comment by Jason-Morcos** [comment]
@chw-webops can you try the latest beta and let us know if the issue persists in the beta as well? 

https://github.com/Sequel-Ace/Sequel-Ace/releases

**Comment by chw-webops** [comment]
@Jason-Morcos I installed the beta v3.1.1 (build 3015), and still experienced the same issue.

**Comment by Jason-Morcos** [comment]
Can you include info about your MySQL instance (at least the version), anything that might make your setup unique, and a screenshot of your connection settings with private info blurred out?

**Comment by chw-webops** [comment]
@Jason-Morcos I've included screenshots of the connection screen and a partial view of my local SSH config.

![sequel-ace-beta-SS02](https://user-images.githubusercontent.com/51791847/105080894-779b6e00-5a5f-11eb-8a23-71c4435d2972.png)
![sequel-ace-beta-SS01](https://user-images.githubusercontent.com/51791847/105080895-78340480-5a5f-11eb-8510-a3a42478ca90.png)

**Comment by Jason-Morcos** [comment]
Ahhh so you're using a custom SSH config in which you css with the server alive settings. Does it "quickly lose connection" if you don't use your custom local SSH config and never to the default ssh config?

**Comment by FaimMedia** [comment]
I have the same problem, since Sequel Ace 3.1 is much more frequent, normally when my MacBook went to sleep this wouldn't occur, now every tab seems to close it's connection and brings-up the reconnect dialog, even if the network connection is already restored and reachable. It hangs for about 20 seconds when selecting another table or opening a new empty tab, when trying to press 'reconnect', the same happens again. I need to close all tabs and create a new connection. When doing that it will also occasionally crash.

**Comment by chw-webops** [comment]
@Jason-Morcos My local SSH settings didn't change the result.

**Comment by octlabs** [comment]
Keeping SSH connections upright got worse and worse for me since the last updates. Before version 3.1 there were no problems. My settings are the same as for years.
Up to version 3.1.1 / 3016 the connection was not restored if it was not active for a few hours. Reconnect was not possible.
But now with 3.1.1 there were problems after just a few minutes that lead to the entire program being blocked by a modal with a permanent error message.

**Comment by Kaspik** [comment]
We are working on it in separate issue as well - https://github.com/Sequel-Ace/Sequel-Ace/issues/658

**Comment by olimortimer** [comment]
I'm also experiencing the same issue.  I recently moved from Sequel Pro to Sequel Ace, with the same connections / setup, yet Sequel Pro didn't have this issue and kept the connection alive.  Happy to provide any info required.

**Comment by Kaspik** [comment]
3.3.0 has been released with major changes in it's hierarchy and structure. Could you please try that? Thanks!

**Comment by octlabs** [comment]
> 3.3.0 has been released with major changes in it's hierarchy and structure. Could you please try that? Thanks!

Still the same issue on Version 3.3.1 Build 3028 also no reconnect, sorry!

**Comment by Kaspik** [comment]
Allright, thanks for checking. Next major change needs to be the connection stuff.

**Comment by Jason-Morcos** [comment]
I believe this is a duplicate of of #658 as @Kaspik mentioned above. Let's consolidate discussion to there.
